task30/src/test/java/by/tut/tests/ReturnObjectFromTable.java

```diff
@@ -0,0 +1,115 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Test(groups = "task40")                                                                                        //Point9
+public class ReturnObjectFromTable {
+    private WebDriver driver;
+    private static final String URL = "https://www.seleniumeasy.com/test/table-sort-search-demo.html";
+    private By nextButton = By.id("example_next");
+    private By lastPageElement = By.cssSelector("div#example_paginate>span>:last-child");
+    private By firstPageElement = By.cssSelector("div#example_paginate>span>:first-child");
+    private By rowElement = By.cssSelector("tbody>tr");
+    private By nameElement = By.cssSelector("td:nth-child(1)");
+    private By positionElement = By.cssSelector("td:nth-child(2)");
+    private By officeElement = By.cssSelector("td:nth-child(3)");
+    private By ageElement = By.cssSelector("td:nth-child(4)");
+    private By salaryElement = By.cssSelector("td:nth-child(6)");
+
+    @BeforeMethod
+    public void setUp() {
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS); //POINT 2
+        driver.manage().window().maximize();
+        driver.get(URL);
+    }
+
+    @AfterMethod
+    public void closeWebDriver() {
+        driver.quit();
+    }
+
+    @Test
+    public void callGetDataFromTabelTestMethod() {
+        int age = 20;
+        int salary = 675000;
+        getDataFromTabelTest(age, salary);
+    }
+
+    public List<EmployeeTableModel> getDataFromTabelTest(int age, int salary) {
+        int lastPage = Integer.parseInt(driver.findElement(lastPageElement).getText());
+        int firstPage = Integer.parseInt(driver.findElement(firstPageElement).getText());
+        List<EmployeeTableModel> tableModelArrayList = new ArrayList<>();
+
+        Predicate<WebElement> isAgeOK = row -> Integer.parseInt(row.findElement(ageElement).getText()) > age ;
+        Predicate<WebElement> isSalaryOK = row -> Integer.parseInt(row.findElement(salaryElement).getAttribute("data-order")) > salary;
+
+        IntStream.rangeClosed(firstPage, lastPage).forEach(page-> {
+            List < WebElement > rowList = driver.findElements(rowElement)
+                    .stream()
+                    .filter(row -> isAgeOK.test(row) && isSalaryOK.test(row))
+                    .collect(Collectors.toList());
+
+            rowList.forEach((element) -> {
+                EmployeeTableModel employee = new EmployeeTableModel();
+
+                employee.setName(element.findElement(nameElement).getText());
+                employee.setPosition(element.findElement(positionElement).getText());
+                employee.setOffice(element.findElement(officeElement).getText());
+
+                tableModelArrayList.add(employee);
+            });
```

Comment:
Let's move this part into map() method in expression above between filter and collect

Reply:
done